### DISCUSSION TITLE
DEX Exchange Rate

### DISCUSSION THREAD

**Discussion by bacchist** [AcalaNetwork/Acala#914] [6. Swap (DEX)]
Using the typical naive liquidity pool model, the exchange rate is a function of the balance of tokens in the pool. As the price on external markets deviates from the price represented by the liquidity pool, market participants can find arbitrage opportunities and the price will gradually rationalize and align generally with the broader market.

The problem with this approach is that it results in impermanent loss suffered by liquidity providers. The economic profit available to arbitragers comes at the expense of liquidity providers. The extracted value may potentially never return to the platform. These inefficiencies are expensive.

The exchange rate should be derived from external data feeds, in order to limit exposure to arbitrage while maintaining an accurate price. Combined with the favorable fee structure, mitigating impermanent loss in this way will help Acala offer an unparalleled value proposition to liquidity providers.

**Comment by georgesdib** [comment]
What's your proposal though? Indeed a constant product AMM suffers from Impermanent Loss, but you have guaranteed liquidity all the way.
If you want to take the price from an external oracle, who would provide liquidity?

**Reply by bacchist** [reply]
The same liquidity providers would provide liquidity. Almost everything would stay the same, as far as I can tell. The only difference would be that the exchange rate would adjust in response to external markets, instead of waiting for arbitragers to come along and change it for an exorbitant fee.

The return on investment for LPs will be so much better than other LP/yield farming options, it's hard to imagine the pools being under-capitalized.

**Reply by georgesdib** [reply]
That does not work, say you have tokens A and B with exchange rate of 1. LP provides 100 A and 100 B, a buyer of A comes and purchases the full 100 of A in exchange of 100 of B (assume price remains unchanged externally), the pool now contains 0A and 200B. In this case, the pool no longer has liquidity to provide.
Hence why the design is to have a function that asymptotically tends to infinity on both extremes of prices to make sure this never happens.
Also obviously, your solution relies on a reliable external price source being available which for small coins is usually not the case.

**Reply by bacchist** [reply]
Would it be possible to have a bonding curve that is influenced by how much liquidity is in the pool? Maybe starting with a feed price when >100% of some target level, and bending towards the implicit price as it drops below that?

Good point that we can't always have a good idea of external price, particularly with less popular tokens. I'm mainly concerned with situations where LPs should reasonably have access to the same information that arbitragers are using.

**Reply by georgesdib** [reply]
Yeah so in a way a bit similar to what curve.fi does for stablecoins but around the oracle price.
So basically, curve.fi uses a bonding curve that is flat around 1 given that is what you would expect the relative price of 2 stablecoins to be.
Here you would use a similar function but around the oracle fed price, so would be dynamic somehow. Not too sure if it works but looks like something promising :) nice!

**Comment by xlc** [comment]
The goal of Acala DEX is not build the best DEX possible. We want something simple, robust and efficient.

Acala is a DeFi platform and Acala DEX won't be the only DEX in our platform. We are currently talking with multiple teams develop DEX / migrate & deploy their DEX to Acala.

If you have some good idea, I will encourage to build it and maybe get it deployed on Acala.

BTW your idea is very similar to DODO https://dodoex.io

**Reply by bacchist** [reply]
That's interesting. I hadn't even considered multiple DEX.

DODO looks pretty cool. I can't believe how much activity there is in this space.
